src/qiboconnection/util.py

```diff
@@ -132,6 +136,32 @@ def unzip(zipped_list: List[Tuple[Any, Any]]):
     return tuple(zip(*zipped_list))
 
 
+def compress_any(any_obj, encoding="utf-8") -> dict:
+    """
+    Transforms any json-serializable object into a compressed string.
+    :param any_obj: object to compress
+    :param encoding: encoding to use for the byte representation
+    :return:
+    """
+
+    encoded_data = json.dumps(any_obj).encode(encoding)
+    compressed_data = base64.b64encode(gzip.compress(encoded_data)).decode()
+    return {"data": compressed_data, "encoding": encoding, "compression": "gzip"}
+
+
+def decompress_any(compressed: str) -> dict:
+    """
+    Decompresses a compressed string into its original datatype.
+    :param compressed: compressed data containing a json to extract a dictionary from
+    :return:
+    """
+
+    data_bin = base64.urlsafe_b64decode(compressed)
+    data = json.loads(gzip.decompress(data_bin))
+
+    return data
```

Comment:
Nitpick, consistency.

You do "return data" here, but "return {blablabla}" on the previous "compress_any" method.

Reply:
Hmm I see. I thought about it, and discarted it because I intuitively thought that for decrompressing you would not need the full "object" (dict in this case), only the compressed bit. 
But I see now that every occurrence of it is used like`decompress_any(blabla["data"})`.

Would it be a good idea to make the signature of `decompress_any` match the structure of our objects? We are currently not using the info, but thats where it could come in handy:

```python
def compress_any(any_obj: str, ...):  # -> {"data": "...", "encoding":"utf-8","compression": "gzip"}
    ...


def decompress_any(data: str,  **kwawrgs):
    ...
```

This makes space for, if we start adding more values that actuate on the parsing steps (compression if it gets relevan, version, etc), we could append them to the explicit signature of decompress any. 


```python

def decompress_any(data: str,  version: str, encoding: str, **kwargs):
    ...
```



And therefore `decompress_any(**compress_any(any_obj))` should work as one would expect.